src/NuGet.Core/NuGet.ProjectModel/ToolPathResolver.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace NuGet.ProjectModel
+{
+    public class ToolPathResolver
+    {
+        private readonly string _packagesDirectory;
+
+        public ToolPathResolver(string packagesDirectory)
+        {
+            _packagesDirectory = packagesDirectory;
+        }
+
+        public string GetLockFilePath(string packageId, SemanticVersion version, NuGetFramework framework)
+        {
+            return Path.Combine(
+                _packagesDirectory,
+                ".tools",
```

Comment:
That's good feedback, but shouldn't block this change. We can add it to the pack work